app/config/AppConfig.scala

```diff
@@ -57,6 +60,12 @@ class AppConfigImpl @Inject() (val config: Configuration, servicesConfig: Servic
   val reportAProblemNonJSUrl: String = s"$contactBaseUrl/contact/problem_reports_nonjs?service=$serviceIdentifier"
   val languageMap: Map[String, Lang] = ListMap("english" -> Lang("en"), "cymraeg" -> Lang("cy"))
 
+
+  private def requestUri(implicit request: RequestHeader) = ContinueUrl(host + request.uri).encodedUrl
```

Comment:
I would keep only the data from configuration here and leave the final construction of the url to its use in the main_layout where you have the request available , this applies to requestUri() and feedbackUrl()